docs: add resource recommendation for charm deployment (#221)
canonical/github-runner-image-builder-operator: app/doc/tutorial/quick-start.md

Reviewer: Nit: Why are there two sets of docs in this repo?

Author: i think the root is for the charm/operator (hence the docs are more on how to deploy or operate the charm). meanwhile the /app is for the actual image-building app that's being charmed. as to why the docs are separated im not sure as it's already like this when i discovered it haha

Reviewer: Ah okay, definitely out of scope for this PR but if we can reduce the documentation duplication in this repo, that would be a very nice improvement :)

- Working [OpenStack environment](https://microstack.run/docs/single-node)
- A `clouds.yaml` configuration with the OpenStack environment

### Recommended resources for deployment

> For reliable image builds, use a recommended flavor for build VMs with:

Reviewer: Maybe this comment is mostly to test my understanding here. But we tell users to use a "recommended flavor". As far as I can tell, we don't actually recommend any flavors in the documentation. 

Unless the "flavor" refers to the resources (2 CPUs, 8 GB RAM, 20 GB disk)? I appreciate any clarity that you provide :)

Author: hi erin yes the flavor refers to any flavor that satisfy the resource requirements!

Reviewer: Great, thanks for clarifying! So perhaps we rephrase:
```suggestion
For reliable image builds, use the following recommended flavor for build VMs:
```
That way it's clearer that the resources map directly to the flavor.
